be/src/exec/vectorized/olap_scan_node.cpp

```diff
@@ -304,13 +304,9 @@ void OlapScanNode::_scanner_thread(TabletScanner* scanner) {
             _close_pending_scanners();
         }
     } else {
-        if (scanner != nullptr) {
-            scanner->close(_runtime_state);
-            _closed_scanners.fetch_add(1, std::memory_order_release);
-            _close_pending_scanners();
-        } else {
-            _close_pending_scanners();
-        }
+        scanner->close(_runtime_state);
```

Comment:
why change this logic？

Reply:
> why change this logic？

tscancode reports an error `Null - checking [scanner] suggests that it may be null, but it has already been dereferenced at line 242.` After my preliminary analysis, it seems that `scanner` cannot be null when executing this code path